ATM Challenge

redleo007 opened the pull request:
I had corrected all the errors and coded as per the requirements of a ATM machine where single user or multiple users can credit and debit money based on their needs .

redleo007 committed:
ATMChallenge

I had corrected all the errors and coded as per the requirements of a ATM machine where single user or multiple users can credit and debit money based on their needs .

santhanam87 commented on src/app/_training/condition.ts:
```diff
@@ -28,25 +8,73 @@ export function shop() {
+  userID:string;
+  balance:number;
+}
+const users: Record<string, accounts>={
+  user1: {userID:"Leo",balance:3000},
+  user2: {userID:"Jeevan",balance:7000},
+  user3: {userID:"Sanjay",balance:2000}
+};
```
Have this as a array and see how you can update the below solution.

redleo007 replied:
I had changed it into array of objects

redleo007 committed:
Changes in ATM

I had changed the objects as list which is array of objects and also used for loop to find user_ID then conditions are already corrected

santhanam87 commented on src/app/_training/condition.ts:
```diff
@@ -27,26 +7,95 @@ export function shop() {
+   {userID:"Jeevan",balance:7000},
+   {userID:"Sanjay",balance:2000}
+];
+function findUserByName(userID: string) {
+  for (const accounts of users) {
+      if (accounts.userID === userID) {
+          return accounts;
+      }
+  }
+  return null;
+}
```
Could you please update this to simple for, for of is a good one, but I like you to understand how the index and other parts of for loops works.

redleo007 replied:
I have updated it into simple for loop

redleo007 committed:
conversion into simple for loop

I had converted it into simple for loop using array's length
